core/src/main/java/org/apache/calcite/rex/RexExecutable.java

```diff
@@ -102,6 +102,38 @@ public void reduce(RexBuilder rexBuilder, List<RexNode> constExps,
     Hook.EXPRESSION_REDUCER.run(Pair.of(code, values));
   }
 
+  /**
+   * Same as {@link #reduce(RexBuilder, List, List)} but with a flag to throw on failure.
+   * <p>Catch the exception to return null for SAFE_ functions.</p>
```

Comment:
This was because you had a comment saying public methods shouldnt just be changed, but a new one should be written and the old deprecated -- maybe that doesn't apply here?

Reply:
You can usually move the method body into the new method (with the extra parameter) and have the old (deprecated) method call the new method.

If the old method is deprecated you should change all calls to use the new method, and remove its javadoc. Thus the new method's javadoc wouldn't be 'Same as (old method)'.